tests/test_core.py

```diff
@@ -8,15 +10,63 @@ class test_Interpolator(unittest.TestCase):
 
     def test_init(self):
         """Test creation from data."""
-        raise NotImplementedError()
+
+        #test construction with tuple input
+        f = relentless.core.Interpolator(x=(-1,0,1), y=(-2,0,2))
+        self.assertEqual(f.domain, (-1,1))
+
+        #test construction with list input
+        f = relentless.core.Interpolator(x=[-1,0,1], y=[-2,0,2])
+        self.assertEqual(f.domain, (-1,1))
+
+        #test construction with numpy array input
+        f = relentless.core.Interpolator(x=np.array([-1,0,1]),
+                                         y=np.array([-2,0,2]))
+        self.assertEqual(f.domain, (-1,1))
+
+        #test construction with mixed input
+        f = relentless.core.Interpolator(x=[-1,0,1], y=(-2,0,2))
+        self.assertEqual(f.domain, (-1,1))
+
+        #test construction with scalar input
+        with self.assertRaises(ValueError):
+            f = relentless.core.Interpolator(x=1, y=2)
+
+        #test construction with 2d-array input
+        with self.assertRaises(ValueError):
+            f = relentless.core.Interpolator(x=np.array([[-1,0,1], [-2,2,4]]),
+                                             y=np.array([[-1,0,1], [-2,2,4]]))
+
```

Comment:
Can you add a test where x and y are both 1d arrays but do not have the same length?